dist/common/scripts/scylla_util.py

```diff
@@ -351,6 +351,199 @@ def user_data(self):
         except urllib.error.HTTPError:  # empty user-data
             return ""
 
+class azure_instance:
+    """Describe several aspects of the current Azure instance"""
+
+    EPHEMERAL = "ephemeral"
+    ROOT = "root"
+    GETTING_STARTED_URL = "http://www.scylladb.com/doc/getting-started-azure/"
+    META_DATA_BASE_URL = "http://169.254.169.254/1.0/meta-data/instance/"
+    ENDPOINT_SNITCH = "GossipingPropertyFileSnitch"
+
+    def __init__(self):
+        self.__type = None
+        self.__cpu = None
+        self.__location = None
+        self.__zone = None
+        self.__memoryGB = None
+        self.__nvmeDiskCount = None
+        self.__firstNvmeSize = None
+        self.__osDisks = None
+
+    @staticmethod
+    def is_azure_instance():
+        """Check if it's Azure instance via DNS lookup to metadata server."""
+        try:
+            addrlist = socket.getaddrinfo('metadata.azure.internal', 80)
+        except socket.gaierror:
+            return False
+        for res in addrlist:
+            af, socktype, proto, canonname, sa = res
+            if af == socket.AF_INET:
+                addr, port = sa
+                if addr == "169.254.169.254":
+                    return True
```

Comment:
Why this check? It seems unnecessary, and potentially breaking for someone trying to emulate the API.

Reply:
this is a good question
the idea if I remember it right was to make sure it's an internet socket and the IP resolved is the one we expect (and that is set by MS Azure folks (and it basically the same for all clouds, but dns name is different))

how do you expect this to be @avikivity to fit your idea? (since I don't feel like relaxing it, resp. don't know how to relax this check in smarter way, sorry :-( )

Reply:
note that this DNS is in local /etc/hosts , so the request shouldn't really involve network

Reply:
By relaxing the check I meant removing the whole thing. Why do you care if it's an inet address or what the address is? If you can connect to that dns name, that's enough.

Reply:
okay, removing it then

Reply:
It doesn't look removed.